### PR TITLE
Shows only the latest attempts of canceled scheduled payment

### PR DESCRIPTION
**Pull request by FadhlanR**
Ticket: CS-5304

**Before**
Screenshot 2023-03-03 at 21 31 38

**After**
Screenshot 2023-03-03 at 21 28 31

### CODE REVIEW

**Comment by jurgenwerk** [comment]
For historic transparency maybe it be better to still display failed payments? And to only show the last one as canceled?

**Comment by FadhlanR** [comment]
@jurgenwerk I have implemented your suggestion. My idea is to add a new field `isCanceled` field in the scheduled payment attempt object in the safe tools.

Screenshot 2023-03-06 at 21 53 57

File: packages/safe-tools-client/app/services/scheduled-payments.ts [hunk 1]
```diff
@@ -178,7 +179,26 @@ export default class ScheduledPaymentsService extends Service {
       'GET'
     );
 
-    return this.deserializeScheduledPaymentAttemptResponse(response);
+    const scheduledPaymentAttempts =
+      this.deserializeScheduledPaymentAttemptResponse(response);
+    const isAlreadyFoundTheLattestAttempt: Record<string, boolean> = {};
```
Comment (lukemelia) [inline packages/safe-tools-client/app/services/scheduled-payments.ts:184]: Maybe a better name would be `previouslySeenPayments`?

**Review by lukemelia — CHANGES_REQUESTED** [review]
Naming needs improvement

**Comment by jurgenwerk** [comment]
Maybe all this can be reduced to:

```hbs
{{if (eq paymentAttempt.status 'failed')}}
  {{#if (and scheduledPaymentAttempt.scheduledPayment.isCanceled  (eq scheduledPaymentAttempt.id scheduledPaymentAttempt.scheduledPayment.lastScheduledPaymentAttemptId)}}
    Canceled
  {{else}}
    Failed
  {{/if}}
{{/if}}
```

**Comment by FadhlanR** [comment]
@jurgenwerk I didn't realize we had that field, thanks for the suggestion. I've updated the code.

**Pull request merged**